workflow.swift

```diff
@@ -56,14 +69,15 @@ wait(ready) {
         printf("swift: The launched application did not succeed.");
     }
 
-    program2 = "stage_write/stage_write";
-    arguments2 = split("heat.bp staged.bp %s \"\" MPI \"\"" % rmethod , " ");
-    printf("size: %i", size(arguments2));
-    printf("swift: launching: %s", program2);
-    exit_code2 = @par=swproc launch(program2, arguments2);
-    printf("swift: received exit code: %d", exit_code2);
-    if (exit_code2 != 0)
+    if(rmethod == "BP")
     {
-        printf("swift: The launched application did not succeed.");
+        wait (exit_code1)
```

Comment:
So if I understand correctly, stage_write must wait for heat_transfer to *finish*, and not just start. Would you know when `launch` returns? After the application is **launched**, or after the application **completes**?